public/apps/configuration/panels/audit-logging/audit-logging-general-settings.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useEffect, useState, useCallback } from 'react';
+import {
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPageHeader,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+  EuiGlobalToastList,
+} from '@elastic/eui';
+import { cloneDeep, set } from 'lodash';
+import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
+import { AppDependencies } from '../../../types';
+import { EditSettingGroup } from './edit-setting-group';
+import { SETTING_GROUPS } from './constants';
+import { API_ENDPOINT_AUDITLOGGING } from '../../constants';
+import { updateAuditLogging } from '../../utils/audit-logging-view-utils';
+
+export function AuditLoggingEditGeneralSetting(props: AppDependencies) {
+  const [config, setConfig] = useState<object>({});
+  const [editConfig, setEditConfig] = useState<object>({});
```

Comment:
Confusing name editConfig to me. Why there are both config and editConfig? I didn't see any section on the screenshot says edit configuration.

Reply:
editConfig is to store the modified configuration.
config is the initial configuration and is used if the user clicks cancel after making some unsubmitted changes.

Reply:
"Cancel" button does not navigate to the view page?

Reply:
Good catch. After checking the behavior of related work, it indeed goes back to view page. Just updated and also removed the config variable which was used to remember the old value.